### DISCUSSION TITLE
Leverage custom context during schema creation?

### DISCUSSION THREAD

**Discussion by scott-deboy** [Netflix/dgs-framework#222] [Ideas]
Would it be possible to expose the custom context to DgsTypeDefinitionRegistry and DgsCodeRegistry implementations for use during schema creation?

**Comment by paulbakker** [comment]
I don't think so - the context is per query/request, while the `DgsTypeDefinitionRegistry` is created once, during initialization of the application.

**Comment by scott-deboy** [comment]
I agree, it wouldn't be for one-time initialization - but I'm looking at performing schema regen/rewriting at runtime as needed based on a custom SchemaReloadIndicator - in those cases having the context would be really helpful (particularly the web request).
